py_neuromodulation/nm_generator.py

```diff
@@ -6,7 +6,7 @@
 def raw_data_generator(
     data: np.ndarray,
     settings: dict,
-    sfreq: int,
+    sfreq: int | float, # Added float, consistent with rest of sfreq variables
```

Comment:
Yes, great!

Reply:
Actually I wanted to ask about this. Sfreq is in many places allowed to be both `int` or `float`, but then in other places it is allowed only as `int` or coerced to `int`.  So, at the time of feature calculation, is the sampling frequency always supposed to end up as a whole number? Because in that case we can handle it at the settings validation level.

Reply:
Thanks for addressing that point, in general it would be great to always the same sfreq value being int or float. So it's great that you added here the float type.
During data retrieval in the nm_generator the float type needs to be casted then into int for correct indexing.

Reply:
OK, so this has been resolved in one of the new commits I added that basically changes all types that were "int|float" to just "float", because actually for the type hinting standard they are exactly the same. Float implies Int basically.

Also, Python does not really know about float or int types like other languages do, it just interprets the numbers as it goes so there is no danger of mixing low level numeric representations.

So after that, now sfreq is basically passed around as a real number until it hits the DataProcessor constructor, at which point it bifurcates into `sfreq_raw` (which is truncated by integer division) and `_sfreq_raw_orig` which is kept as a copy of the original value. Does that check out?